### PR TITLE
Pyam > 1.0

### PR DESCRIPTION
**Pull request by znicholls**
# Pull request

Please confirm that this pull request has done the following:

- [x] Tests added (N/A)
- [x] Documentation added (where applicable) (N/A)
- [x] Example added (either to an existing notebook or as a new notebook, where applicable) (N/A)
- [x] Description in ``CHANGELOG.rst`` added
- [x] #138 

## Adding to CHANGELOG.rst

Please add a single line in the changelog notes similar to one of the following:

```
- (`#XX <https://github.com/GranthamImperial/silicone/pull/XX>`_) Added feature which does something
- (`#XX <https://github.com/GranthamImperial/silicone/pull/XX>`_) Fixed bug identified in (`#YY <https://github.com/GranthamImperial/silicone/issues/YY>`_)
```

### CODE REVIEW

File: src/silicone/multiple_infillers/infill_all_required_emissions_for_openscm.py [hunk 1]
```diff
@@ -273,58 +287,42 @@ def _perform_crunch_and_check(
...
-        mdf = to_fill_orig.filter(model=model, variable=leaders + required_variables)
-        for scenario in mdf.scenario:
-            msdf = mdf.filter(scenario=scenario)
-            msdf_filled = to_fill.filter(
-                model=model, scenario=scenario, variable=msdf["variable"].unique()
-            )
+    orig_ts = to_fill_orig.timeseries()
+    common_times = check_ts.columns.intersection(orig_ts.columns)
+    if not common_times.empty:
+        check_ts, orig_ts = check_ts.align(orig_ts, join="right")
```
Comment (Rlamboll) [inline src/silicone/multiple_infillers/infill_all_required_emissions_for_openscm.py:318]: This is a much more efficient way to do that!

File: CHANGELOG.rst [hunk 2]
```diff
@@ -3,14 +3,24 @@ Changelog
...
+------
 
 Added
 ~~~~~
+- (`#139 <https://github.com/GranthamImperial/silicone/pull/139>`_) Support for pyam-iamc>1.0
 - (`#135 <https://github.com/GranthamImperial/silicone/pull/135>`_) Added html documentation of the time projectors
 
+Changed
+~~~~~~~
+- (`#138 <https://github.com/GranthamImperial/silicone/pull/138>`_) Drop Python3.6 support
```
Comment (Rlamboll) [inline CHANGELOG.rst]: Feels like a "remove" to me

File: CHANGELOG.rst [hunk 3]
```diff
@@ -3,14 +3,24 @@ Changelog
...
+
+Fixed
+~~~~~
+- (`#138 <https://github.com/GranthamImperial/silicone/pull/138>`_) Improve speed of :func:`silicone.multiple_infillers.infill_all_required_emissions_for_openscm` by removing multiple loops
```
Comment (Rlamboll) [inline CHANGELOG.rst]: I would say this is a change not a fix, as the previous behaviour was inefficient rather than broken.

**Review by Rlamboll — APPROVED** [review]
Why are we dropping support for python 3.6? Otherwise I disagree with the changelog nomenclature but the actual code all looks good to me and this seems a useful thing to speed up! I recommend just doing everything in one go with this merge, as otherwise changelog clashes occur.

**Comment by jkikstra** [comment]
Some short explanation on why 3.6 was dropped is in #138 :
> 
> 
> The CI was failing with Python 3.6 because of an error related to openscm-units so I just dropped Python3.6 support (none of the scmdata stack supports 3.6 anymore).

Thanks for this great speed-up @znicholls 👍

File: src/silicone/multiple_infillers/infill_all_required_emissions_for_openscm.py (hunk 1 above)
Comment (znicholls) [reply src/silicone/multiple_infillers/infill_all_required_emissions_for_openscm.py:318]: 👍

File: CHANGELOG.rst (hunk 3 above)
Comment (znicholls) [reply CHANGELOG.rst]: No worries I can move it.

For completeness, the reason I did it here is that this is that our CHANGELOG is based on https://keepachangelog.com/en/1.0.0/. This isn't a breaking change so it didn't feel like it belonged in `Changed`, but it's true that there's not a perfect category for it.

**Comment by znicholls** [comment]
> Why are we dropping support for python 3.6?

OpenSCM-Units relies on importlib and the part of importlib it uses doesn't exist in Python 3.6. I don't think we lose anything by dropping 3.6 support?

**Comment by znicholls** [comment]
@Rlamboll if you're happy with the updates can you merge and then release a new version please (according to super strict semantic versioning we should probably go 2.0 cause we dropped Python 3.6 but I think everything is really just a bug fix so 1.1.1 or 1.2.0 are also justifiable).

**Comment by znicholls** [comment]
Hi @Rlamboll sorry just to clarify, should I do a new release or will you?

**Comment by Rlamboll** [comment]
I started drafting it already but didn't commit. Trying to work out what to say about python 3.6 - although we don't officially support it, there's nothing in this update that specifically breaks it either, it's just lack of commitment based on the requirements of other packages. I think this doesn't constitute a new whole integer version since none of the interface has changed.

**Comment by znicholls** [comment]
Sounds good, I'm happy with whatever you go with

**Comment by Rlamboll** [comment]
OK, done a release

**Pull request merged**